### PR TITLE
Remove angular-ui-router and ui-router-extras, which have been replaced by packages that provide their own types

### PR DESCRIPTION
**Pull request by ghost**
If this is merged I will run:
* `npm deprecate @types/angular-ui-router "'@types/angular-ui-router' is now '@uirouter/angularjs', which provides its own types."`
* `npm deprecate @types/ui-router-extras "'ui-router-extras' is now part of various 'uirouter' plugins. See https://github.com/christopherthielen/ui-router-extras`

CC @michelsalib, @matiishyn, @mikehaas763, @mputters, @marcel-k, @LaserUnicorns

### CODE REVIEW

File: types/angular-breadcrumb/index.d.ts [hunk 1]
```diff
@@ -4,8 +4,6 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
-/// <reference types="angular-ui-router" />
```
Comment (andy-ms) [inline types/angular-breadcrumb/index.d.ts]: Didn't seem to use the dependency.

**Comment by mikehaas763** [comment]
I didn’t realize that happened, let me pull down the new types real quick and see if my large project blows up

**Comment by mikehaas763** [comment]
Does this mean that essentially we're forced to use at least the oldest ui router release available in @uirouter/angularjs? Rather than deprecating this @types packages can we just use the version feature where it's only to be used for say 0.x.x versions of ui router. We're on a really old version of uirouter and will probably never upgrade. Do I need to continue using the definitely typed types?

Removing @types/angular-ui-router from my project and then installing @uirouter/angularjs results in "ERROR in [at-loader] ../../node_modules/@types/angular/jqlite.d.ts:811:5 TS2374: Duplicate string index signature." Do I need to open an issue with the ui router project?

**Comment by andy-ms** [comment]
@mikehaas763 I'm guessing that error is from `@types/angular` (GH must have removed that part from your comment since it looks like a username). The latest version of that doesn't define any string indexer at line 811 though, so maybe you need to update that?

Deprecating this package won't prevent you from continuing to use it if you're still on an `angular-ui-router`, it will just result in a message when you install it reminding you that the newer version exists.

**Comment by mikehaas763** [comment]
> Deprecating this package won't prevent you from continuing to use it

I should have worded my original question more clearly. What I'm after is, can we still iterate on the @types/angular-ui-router types once deprecated?

**Comment by andy-ms** [comment]
If you're still planning to make updates to these types I guess they shouldn't be removed from DefinitelyTyped just yet -- we could manually deprecate but leave them here for now.

**Comment by mhegazy** [comment]
@andy-ms what is the plan? are we keeping them or removing them?

**Comment by andy-ms** [comment]
I guess we should leave them for now. If we don't see activity after a few months we can reconsider removing them.

**Pull request closed without merging**